File: architecture/data/src/main/java/com/pacific/data/SchedulersUtil.java
package com.pacific.data;

import android.os.Looper;

import io.reactivex.android.MainThreadDisposable;
import io.reactivex.android.schedulers.AndroidSchedulers;

public class SchedulersUtil {
    private SchedulersUtil() {
        throw new UnsupportedOperationException();
    }

    public static boolean isMainThread() {
        try {
            return Looper.myLooper() == Looper.getMainLooper();
        } catch (Exception e) {
            return true;// Cover for tests
        }
    }

    public static void verifyMainThread() {
        if (BuildConfig.DEBUG) {
            return;// Cover for tests
        }
        MainThreadDisposable.verifyMainThread();
    }

    public static void verifyWorkThread() {
        if (BuildConfig.DEBUG) {
            return;// Cover for tests
        }
        if (isMainThread()) {
            throw new UnsupportedOperationException("Can't run in Main thread");
        }
    }

    public static void postToMainThread(Runnable runnable) {
        AndroidSchedulers.mainThread().scheduleDirect(runnable);
    }
}

File: architecture/arch-kotlin/src/main/java/com/pacific/arch/kotlin/rx/SchedulersUtil.kt
package com.pacific.arch.kotlin.rx

import android.os.Looper
import com.pacific.arch.kotlin.BuildConfig
import io.reactivex.android.MainThreadDisposable
import io.reactivex.android.schedulers.AndroidSchedulers

fun isMainThread(): Boolean {
    return try {
        Looper.myLooper() == Looper.getMainLooper()
    } catch (e: Exception) {
        true// Cover for tests
    }
}

fun verifyMainThread() {
    if (BuildConfig.DEBUG) {
        return // Cover for tests
    }
    MainThreadDisposable.verifyMainThread()
}

fun verifyWorkThread() {
    if (BuildConfig.DEBUG) {
        return // Cover for tests
    }
    if (isMainThread()) {
        throw UnsupportedOperationException("Can't run in MainThread")
    }
}

fun postToMainThread(runnable: Runnable) {
    AndroidSchedulers.mainThread().scheduleDirect(runnable)
}